src/mongoose_hooks.erl

```diff
@@ -513,8 +519,11 @@ user_sent_keep_alive(HostType, JID) ->
     Packet :: exml:element(),
     Result :: mongoose_acc:t().
 user_send_packet(Acc, From, To, Packet) ->
+    Params = #{from => From, to => To, packet => Packet},
+    Args = [From, To, Packet],
+    ParamsWithLegacyArgs = ejabberd_hooks:add_args(Params, Args),
```

Comment:
Actually this one is one that I'll change again during my c2s rework, because `from`, `to`, and `packet` are already part of the `mongoose_acc`, and so it always felt redundant to have all those parameters. So when they're needed, the handler can do `{From, To, Packet} = mongoose_acc:packet(Acc)` and will get them. So many handlers actually ignore one or another or both... So I don't know if we want to require a migration for the handlers in the next release, and yet another migration for the same handler in the following one, after my c2s work is merged 😕 

@DenysGonchar what do you think?

Reply:
Putting here some notes after conversation with @NelsonVides:

It's proposed to follow this rules to reduce the number of merge conflicts in the future, and prevent collisions between the modules:
- `Acc` should be the source of such data, because some of the handlers may extend the `Packet` with extra fields/attributes, we must ensure that one handler cannot not accidentally override the changes done in another handler. 
- please don't put anything in `Params` except the legacy `Args`.
- in the reworked hook handler functions get `{From, To, Packet}` parameters from the `Acc`.

I believe the same comment is relevant to `rest_user_send_packet` and probably some other send*/filter* hooks.

Reply:
I applied changes to follow these rules. Please check if it is correct.

Reply:
thanks